src/components/products/FilterAndSort.jsx

```diff
@@ -1,5 +1,189 @@
+          onClick={() => filterDispatch({ type: "CLEAR" })}
+        >
+          Clear
+        </button>
+      </div>
+      <div className="my-xs">
+        <label>
+          <input
+            type="radio"
+            name="sorting"
+            checked={filterState.sorting === "lth"}
+            onClick={() =>
+              filterDispatch({ type: "LOW_TO_HIGH", payload: "lth" })
+            }
+          />
+          <small>Price: Low to High</small>
+        </label>
+        <label>
+          <input
+            type="radio"
+            name="sorting"
+            checked={filterState.sorting === "htl"}
+            onClick={() =>
+              filterDispatch({ type: "HIGH_TO_LOW", payload: "htl" })
+            }
+          />
+          <small>Price: High to Low</small>
+        </label>
+      </div>
+      <form className="flex flex-col my-xs">
+        <h5 className="my-xs">Categories</h5>
+        {categories.map(
+          (cat) =>
+            (cat.categoryType === "genre" ||
+              cat.categoryType === "special") && (
+              <label>
+                <input
+                  type="checkbox"
+                  checked={filterState.categoryName.includes(cat.categoryName)}
+                  onClick={() =>
+                    filterDispatch({
+                      type: "SET_CATEGORY",
+                      payload: cat.categoryName,
+                    })
+                  }
+                />
+                <small>{cat.categoryName}</small>
+              </label>
+            )
+        )}
+      </form>
+      <form className="flex my-s">
+        <h5>Price</h5>
+        <label>
+          <input
+            className="mx-s"
+            type="range"
+            min="50"
+            max="2000"
+            value={filterState.maxPrice}
+            onChange={(e) =>
+              filterDispatch({ type: "SET_MAX_PRICE", payload: e.target.value })
+            }
+          />
+          {filterState.maxPrice}
+        </label>
+      </form>
+      <form>
+        <label className="my-xs">
+          <input
+            type="checkbox"
+            checked={filterState.includeOOS}
+            onClick={(e) =>
+              filterDispatch({
+                type: "INCLUDE_OUT_OF_STOCK",
+                payload: e.target.checked,
+              })
+            }
+          />
+          <small>Include Out of Stock</small>
+        </label>
+        <label className="my-xs">
+          <input
+            type="checkbox"
+            checked={filterState.fastDeliveryOnly}
+            onClick={(e) =>
+              filterDispatch({
+                type: "FAST_DELIVERY_ONLY",
+                payload: e.target.checked,
+              })
+            }
+          />
+          <small>Fast Delivery Only</small>
+        </label>
+      </form>
+
+      <form className="flex flex-col my-xs">
+        <h5 className="my-xs">Format</h5>
+        <label for="pb">
```

Comment:
In JSX we need to write `htmlFor` instead of `for` as it is a reserved keyword

Reply:
Thanks for pointing out. Changed to ```htmlFor```